src/ConfigParser/config_parser.cpp

```diff
@@ -0,0 +1,396 @@
+    double nr_exp          = parser_.get<int>("nr_exp");
+    int ntheta_exp         = parser_.get<int>("ntheta_exp");
+    int anisotropic_factor = parser_.get<int>("anisotropic_factor");
+    int divideBy2          = parser_.get<int>("divideBy2");
+
+    // Parse multigrid parameters from command-line arguments
+    FMG_                     = parser_.get<int>("FMG") != 0;
+    FMG_iterations_          = parser_.get<int>("FMG_iterations");
+    const int FMG_cycleValue = parser_.get<int>("FMG_cycle");
+    if (FMG_cycleValue == static_cast<int>(MultigridCycleType::V_CYCLE) ||
+        FMG_cycleValue == static_cast<int>(MultigridCycleType::W_CYCLE) ||
+        FMG_cycleValue == static_cast<int>(MultigridCycleType::F_CYCLE)) {
+        FMG_cycle_ = static_cast<MultigridCycleType>(FMG_cycleValue);
+    }
+    else {
+        throw std::runtime_error("Invalid FMG cycle type.");
+    }
+    const int extrapolationValue = parser_.get<int>("extrapolation");
+    if (extrapolationValue == static_cast<int>(ExtrapolationType::NONE) ||
+        extrapolationValue == static_cast<int>(ExtrapolationType::IMPLICIT_EXTRAPOLATION) ||
+        extrapolationValue == static_cast<int>(ExtrapolationType::IMPLICIT_FULL_GRID_SMOOTHING) ||
+        extrapolationValue == static_cast<int>(ExtrapolationType::COMBINED)) {
+        extrapolation_ = static_cast<ExtrapolationType>(extrapolationValue);
+    }
+    else {
+        throw std::runtime_error("Invalid extrapolation type.");
+    }
+    max_levels_           = parser_.get<int>("maxLevels");
+    pre_smoothing_steps_  = parser_.get<int>("preSmoothingSteps");
+    post_smoothing_steps_ = parser_.get<int>("postSmoothingSteps");
+    const int cycleValue  = parser_.get<int>("multigridCycle");
+    if (cycleValue == static_cast<int>(MultigridCycleType::V_CYCLE) ||
+        cycleValue == static_cast<int>(MultigridCycleType::W_CYCLE) ||
+        cycleValue == static_cast<int>(MultigridCycleType::F_CYCLE)) {
+        multigrid_cycle_ = static_cast<MultigridCycleType>(cycleValue);
+    }
+    else {
+        throw std::runtime_error("Invalid multigrid cycle type.");
+    }
+    max_iterations_     = parser_.get<int>("maxIterations");
+    const int normValue = parser_.get<int>("residualNormType");
+    if (normValue == static_cast<int>(ResidualNormType::EUCLIDEAN) ||
+        normValue == static_cast<int>(ResidualNormType::WEIGHTED_EUCLIDEAN) ||
+        normValue == static_cast<int>(ResidualNormType::INFINITY_NORM)) {
+        residual_norm_type_ = static_cast<ResidualNormType>(normValue);
+    }
+    else {
+        throw std::runtime_error("Invalid residual norm type.");
+    }
+    double absTol       = parser_.get<double>("absoluteTolerance");
+    absolute_tolerance_ = (absTol >= 0) ? std::optional<double>(absTol) : std::nullopt;
+    double relTol       = parser_.get<double>("relativeTolerance");
+    relative_tolerance_ = (relTol >= 0) ? std::optional<double>(relTol) : std::nullopt;
```

Comment:
Condition doesn't match default value. Should the default for the parser be a negative value?

Reply:
Maybe I'm missing something. What happens if you pass `std::nullopt`?